apis/core/v1alpha1/condition.go

```diff
@@ -285,6 +287,8 @@ func ReferenceResolutionSuccess() Condition {
 // unable to resolve the references used in the resource. This could
 // mean that one or more of referred resources do not yet exist, or are not yet
 // Ready.
+//
+// Deprecated: Use ReconcileError.
```

Comment:
Since we are expecting for some resources that they may be in a state of waiting on other resources for a non-trivial period of time, does it make sense to make blocked on reference resolution a reconcile error condition? It feels like it is still successful reconciliation.

Reply:
My feeling is that there's less value in having a distinct condition for references now that we also emit a `CannotResolveResourceReferences` event.

It's also worth noting that `reference.NewAPIResolver` doesn't check the `Ready` condition - so the only reason we'd report a reconcile error would be if you referenced a managed resource that did not actually exist. If this persists for a non-trivial amount of time I'd suspect it was actually a configuration error (i.e. you typoed your reference).